### PR TITLE
[MXNET-730][WIP] Scala test in nightly

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
This is the PR for migrating Scala test to nightly build. The action I applied here is to provide a new function in `Util` that can check which test to run and how many epochs needed to run. All Integration test will run 1 epoch in default and nightly will run more for performance and accuracy checking.

@nswamy @yzhliu @andrewfayres @marcoabreu 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala [hunk 1]
```diff
@@ -42,4 +48,30 @@ object Util {
     }
    if (!success) throw new Exception(s"$url Download failed!")
   }
+
+  /**
+    * This Util is designed to manage the tests in CI
+    * @param name the name of the test
+    * @return runTest and number of epoch
+    */
+  def testManager(name: String) : (Boolean, Int) = {
```
Comment (nswamy) [inline scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: why is this needed ? can this not be controlled through make/maven targets and configurations?

Comment (lanking520) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: Then there will be 10+ lines maven configuration and fragments of checking in the tests to achieve the same functionalities. I believe placing them all together into one place is better to manage

Comment (nswamy) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: you'll need to change everytime a new test it added. Ideally your test should be configurable and you choose what settings you want to run in the target/script.

I don't think it should be controlled through the code here

Comment (lanking520) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: Let's say this way. If someday you want to add a new flag to split the test further for some other use cases, you need to open each test and add that flag there. Assume you have 10 tests in the plate, then 10 duplicated code change will be applied in the tests. This solution, one change would be necessary. Apart from that, this holds the same scenario of downloading files, we create a standard input and output into a single place will definite helps us for future maintenance.

Comment (andrewfayres) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: Agreeing with Naveen here. Convention is that we control this type of behavior through something like environment variables, build parameters, or config files.

One other thing is that we should try to avoid having code which is unique to our CI testing inside the src folder.

Comment (larroy) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: isn't that why we have src/test/scala in the standard layout?

Comment (andrewfayres) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/Util.scala]: Yes, you're right. I meant to say we should avoid having it in src/main folder.

**Comment by lanking520** [comment]
Close this for now as it is not yet settled to an agreement. Will think about a better way and open again.
